src/main/java/org/spongepowered/api/event/network/ClientConnectionEvent.java

```diff
@@ -80,10 +86,22 @@
      * Called after the client authenticates and attempts to login to the
      * server.
      *
-     * <p>Note: This event is fired after #Auth and is NOT async. Any changes 
-     * required for the {@link Player}s {@link Transform} should be done during 
+     * <p>Note: This event is fired after #Auth and is NOT async. Any changes
+     * required for the {@link Player}s {@link Transform} should be done during
      * this event and NOT during #Join.
      * </p>
+     *
+     * <p>If registered {@link BanService} or {@link WhitelistService}
+     * indicates that a player should not be allowed to join
+     * ({@link GameProfile} or {@link InetAddress} has an ban, or is
+     * not on the whitelist), then this event will automatically cancelled by
+     * the {@link Platform#getImplementation() 'game' plugin}, with the proper
```

Comment:
@simon816: Only the Vanilla `/ban` command kicks the player after a ban. Other than that, you can't really detect if it was from a ban.